easybuild/tools/filetools.py

```diff
@@ -604,33 +613,41 @@ def apply_patch(patch_file, dest, fn=None, copy=False, level=None):
     - assume unified diff created with "diff -ru old new"
     """
 
-    if not os.path.isfile(patch_file):
+    if build_option('extended_dry_run'):
+        # skip checking of files in dry run mode
+        patch_filename = os.path.basename(patch_file)
+        dry_run_msg("* applying patch file %s" % patch_filename, silent=build_option('silent'))
+
+    elif not os.path.isfile(patch_file):
         raise EasyBuildError("Can't find patch %s: no such file", patch_file)
-        return
 
-    if fn and not os.path.isfile(fn):
+    elif fn and not os.path.isfile(fn):
         raise EasyBuildError("Can't patch file %s: no such file", fn)
-        return
 
-    if not os.path.isdir(dest):
+    elif not os.path.isdir(dest):
         raise EasyBuildError("Can't patch directory %s: no such directory", dest)
-        return
 
     # copy missing files
     if copy:
-        try:
-            shutil.copy2(patch_file, dest)
-            _log.debug("Copied patch %s to dir %s" % (patch_file, dest))
-            return 'ok'
-        except IOError, err:
-            raise EasyBuildError("Failed to copy %s to dir %s: %s", patch_file, dest, err)
-            return
+        if build_option('extended_dry_run'):
+            dry_run_msg("  %s copied to %s" % (patch_file, dest), silent=build_option('silent'))
+        else:
+            try:
+                shutil.copy2(patch_file, dest)
+                _log.debug("Copied patch %s to dir %s" % (patch_file, dest))
+                return 'ok'
+            except IOError, err:
+                raise EasyBuildError("Failed to copy %s to dir %s: %s", patch_file, dest, err)
 
     # use absolute paths
     apatch = os.path.abspath(patch_file)
     adest = os.path.abspath(dest)
 
-    if not level:
+    patch_level = level
+    if patch_level is None and build_option('extended_dry_run'):
+        patch_level = '<derived>'
```

Comment:
that's skipped already (`run_cmd` is modified to not execute any commands unless forced)

trying to derive the patch level should never be tried though, since that requires the patch file to be available, and to find/read it, etc.